app/code/BVMyBlog/Blog/Controller/Adminhtml/Post/Save.php

```diff
@@ -0,0 +1,120 @@
+
+use BVMyBlog\Blog\Api\BlogRepositoryInterface;
+use BVMyBlog\Blog\Model\Blog;
+use BVMyBlog\Blog\Model\BlogFactory;
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\App\Request\DataPersistorInterface;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Saves data from form
+ */
+class Save extends Action implements HttpPostActionInterface
+{
+    const ADMIN_RESOURCE = 'BVMyBlog_Blog::blog_manage_posts';
+
+    /**
+     * @var mixed $blogFactory
+     */
+    private $blogFactory;
+
+    /**
+     * @var DataPersistorInterface
+     */
+    protected $dataPersistor;
+
+    /**
+     * @var BlogRepositoryInterface
+     */
+    private $blogRepository;
+
+    /**
+     * @param Context $context
+     * @param DataPersistorInterface $dataPersistor
+     * @param BlogFactory $blogFactory
+     * @param BlogRepositoryInterface $blogRepository
+     */
+    public function __construct(
+        Context $context,
+        DataPersistorInterface $dataPersistor,
+        BlogFactory $blogFactory,
+        BlogRepositoryInterface $blogRepository
+    ) {
+        $this->dataPersistor = $dataPersistor;
+        $this->blogFactory = $blogFactory;
+        $this->blogRepository = $blogRepository;
+        parent::__construct($context);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $data = $this->getRequest()->getPostValue();
+
+        if ($data) {
+            if (empty($data['post_id'])) {
+                $data['post_id'] = null;
+            }
+            /** @var Blog $post */
+            $post = $this->blogFactory->create(['data' => $data]);
+            $imgPath = $data['image_path'][0]['url'] ?? '';
+            $data['image_path'] = $imgPath;
+
+            $id = $this->getRequest()->getParam('id');
+            if ($id) {
+                try {
+                    $post = $this->blogRepository->getById($id);
+                } catch (NoSuchEntityException $e) {
+                    $this->messageManager->addErrorMessage(__('This post no longer exists.'));
+                    return $resultRedirect->setPath('*/*/');
+                }
+            }
+            $post->addData($data);
+            try {
+                $this->blogRepository->save($post);
+                $this->messageManager->addSuccessMessage(__('You saved the post.'));
+                $this->dataPersistor->clear('bvmyblog_blog_post');
+                return $this->processBlockReturn($post, $data, $resultRedirect);
+            } catch (CouldNotSaveException $e) {
+                $this->messageManager->addErrorMessage($e->getMessage());
+            } catch (\Exception $e) {
```

Comment:
\Exception should not be used. Use more specific exception. Refer to https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html#5-exceptions